src/coreclr/nativeaot/System.Private.CoreLib/src/System/Exception.CoreRT.cs

```diff
@@ -15,9 +15,16 @@ namespace System
 {
     public partial class Exception
     {
-        // TargetSite is not supported on CoreRT. Because it's likely use is diagnostic logging, returning null (a permitted return value)
-        // seems more useful than throwing a PlatformNotSupportedException here.
-        public MethodBase TargetSite => null;
+        public MethodBase TargetSite
+        {
+            get
+            {
+                if (!HasBeenThrown)
+                    return null;
+
+                return new StackFrame(_corDbgStackTrace[_idxFirstFreeStackTraceEntry - 1], needFileInfo: false).GetMethod();
```

Comment:
```suggestion
                return new StackFrame(_corDbgStackTrace[0], needFileInfo: false).GetMethod();
```
?

Reply:
Yea. I don't know what I was thinking.

Also submitted https://github.com/dotnet/runtime/pull/50275 so that we have test coverage for the bug I almost introduced.